opm/autodiff/FlowMain.hpp

```diff
@@ -255,6 +256,77 @@ namespace Opm
 
 
 
+        /// helper function checking that _one_ coordinate differs by 1
+        bool ijk_are_adj(int i1, int j1, int k1, int i2, int j2, int k2) {
+            return ((i1-i2)*(i1-i2) +
+                    (j1-j2)*(j1-j2) +
+                    (k1-k2)*(k1-k2)) == 1;
+        }
+
+        std::array<int, 3> getIJK(size_t g, const Grid& grid) {
+            using namespace UgGridHelpers;
+            std::array<int, 3> r;
+            const int * dimens = cartDims(grid);
+            size_t m_nx = dimens[0];
+            size_t m_ny = dimens[1];
+            size_t m_nz = dimens[2];
+
+            int i = g % m_nx;  g /= m_nx;
+            int j = g % m_ny;  g /= m_ny;
+            int k = g % m_nz;
+
+            r[0] = i; r[1] = j; r[2] = k;
+            return r;
+        }
+
+        /// checks cartesian adjacency of global indices g1 and g2
+        bool cartesianAdjacent(size_t g1, size_t g2, const Grid& grid) {
```

Comment:
This can alternatively be implemented as:
```C++
auto diff = std::abs(g1 - g2);

const int * dimens = cartDims(grid);
if (diff == 1)
   return true;
if (diff == dimens[0])
   retturn true;
if (diff == dimens[0] * dimens[1])
   return true;

return false;
```

Then the whole (ijk) fencing can be avoided.